### PR TITLE
dockerize front end

### PR DESCRIPTION
**Pull request by vim-usds**
Looking to get some feedback on this. It would be awesome if someone can validate that `docker-compose up` works!

1. Install docker [desktop](https://www.docker.com/products/docker-desktop)
2. open docker desktop
3. fetch and checkout this branch
4. run `docker-compose up`
5. open docker desktop
6. click on containers (left hand side of docker desktop)
6. click to expand `justice40-tool`
7. hover on the `j40-web-container`
8. click on the icon to open in web (right hand side)
9. validate app displays and works

### CODE REVIEW

File: client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM node:14
+
+WORKDIR /app
+
+COPY package*.json ./
+
+RUN yarn install --network-concurrency 2
```
Comment (vim-usds) [inline client/Dockerfile:7]: concurrency reduction was added to have the `install` pass

`npm install` was getting blocked by trussworks
`yarn install` was getting blocked by [lodash](https://github.com/yarnpkg/yarn/issues/6312)

File: client/package.json [hunk 2]
```diff
@@ -9,8 +9,8 @@
   ],
   "license": "CC0-1.0",
   "scripts": {
-    "develop": "gatsby develop",
-    "start": "gatsby develop",
+    "develop": "gatsby develop -H 0.0.0.0",
```
Comment (vim-usds) [inline client/package.json:12]: added to allow docker access to gatsby

File: docker-compose.yml [hunk 3]
```diff
@@ -13,3 +13,9 @@ services:
     environment:
       ENV_FOR_DYNACONF: development
       PYTHONUNBUFFERED: 1
+  j40_web:
```
Comment (vim-usds) [inline docker-compose.yml:16]: Not set on these names, if there are others that are more appropriate we can change. Thx

`j40_web`
`j40_web_container`

**Comment by ginabeena** [comment]
@vim-usds is this documentation also part of your PR or just for the PR description?

**Comment by vim-usds** [comment]
> > @vim-usds is this documentation also part of your PR or just for the PR description?
> 
> Like, also maybe add a:
> 
> * [ ]  vetted that these instructions are repeatable by a team member before merge
> 
> to the description

Thanks Gina!

Yeah so right now it's just part of the ticket. I think it should be documented somewhere. Shelby and I were just talking about documentation and seeing where this would fit in. I think we'll talk about at our next dev sync. Do you have any ideas as to where this should go?

For now I'll put it on the PR and the ticket.

**Comment by ginabeena** [comment]
> > > @vim-usds is this documentation also part of your PR or just for the PR description?
> > 
> > 
> > Like, also maybe add a:
> > 
> > * [ ]   vetted that these instructions are repeatable by a team member before merge
> > 
> > to the description
> 
> Thanks Gina!
> 
> Yeah so right now it's just part of the ticket. I think it should be documented somewhere. Shelby and I were just talking about documentation and seeing where this would fit in. I think we'll talk about at our next dev sync. Do you have any ideas as to where this should go?
> 
> For now I'll put it on the PR and the ticket.

A README.md in the client directory would probably be the place

File: docker-compose.yml (hunk 3 above)
Comment (esfoobar-usds) [reply docker-compose.yml:16]: I personally think the names are good.

**Review by esfoobar-usds — COMMENTED** [review]
I don't see the environment variable handling anywhere here, I'm assuming you haven't pushed?

**Comment by vim-usds** [comment]
> I don't see the environment variable handling anywhere here, I'm assuming you haven't pushed?

Hi @Jorge, just added env variables. Please take a look if you get a chance

**Comment by esfoobar-usds** [comment]
The env vars handling looks good. I just have a question -- where in the Gatsby code are you actually setting the URLs if the env.data_source is local or staging?

File: docker-compose.yml [hunk 4]
```diff
@@ -13,3 +13,11 @@ services:
...
+    image: j40_web
+    container_name: j40_web_container
+    build: ./client
```
Comment (esfoobar-usds) [inline docker-compose.yml:19]: Note: you're missing this here for local access:

```
    volumes:
      - ./data/data-pipeline:/data-pipeline
 ```

**Comment by vim-usds** [comment]
> The env vars handling looks good. I just have a question -- where in the Gatsby code are you actually setting the URLs if the env.data_source is local or staging?

Yeah right!

There's a few files for this:

env.development:
```
GATSBY_DATA_CDN_BASE_URL=https://d3jqyw10j8e7p9.cloudfront.net
GATSBY_DATA_PIPELINE_SCORE_PATH=data-pipeline/data/score
GATSBY_SCORE_DOWNLOAD_FILE_PATH=downloadable/Screening+Tool+Data.zip
GATSBY_MAP_TILES_PATH=tiles
```

env.local:
```
GATSBY_DATA_CDN_BASE_URL=../data
GATSBY_DATA_PIPELINE_SCORE_PATH=data-pipeline/data/score
GATSBY_SCORE_DOWNLOAD_FILE_PATH=downloadable/Screening+Tool+Data.zip
GATSBY_MAP_TILES_PATH=tiles
```

For `env.development`, wondering if it should change to `env.staging`?

And if DATA_CDN_BASE_URL should actually be named DATA_ROOT_DIR since it could be a CDN or it could be local?

**Review by esfoobar-usds — APPROVED** [review]
This works well now. I see some `socket.io` errors on console but they don't seem to be affecting the map.

Great job @vim-usds

**Comment by vim-usds** [comment]
> This works well now. I see some `socket.io` errors on console but they don't seem to be affecting the map.
> 
> Great job @vim-usds

Thank you @esfoobar-usds , couldn't have done it without your support! (and @switzersc-usds too) !

**Pull request merged**